library/analytics/src/main/java/org/cru/godtools/analytics/appsflyer/AppsFlyerAnalyticsService.kt

```diff
@@ -58,26 +75,43 @@ class AppsFlyerAnalyticsService @Inject internal constructor(private val app: Ap
             appsFlyer.trackEvent(app, event.action, emptyMap())
     }
     // endregion Analytics Events
-}
 
-private object GodToolsAppsFlyerConversionListener : AppsFlyerConversionListener {
-    override fun onAppOpenAttribution(data: Map<String, String>?) {
-        data?.map { (key, value) ->
-            Timber.tag(TAG).d("AppsFlyer onAppOpenAttribution Attribute: %s = %s", key, value)
-        }
-    }
+    // region Application.ActivityLifecycleCallbacks
+    private var activeActivity: Activity? by weak()
 
-    override fun onAttributionFailure(error: String?) {
-        Timber.tag(TAG).e("AppsFlyer onAttributionFailure: %s", error)
+    override fun onActivityResumed(activity: Activity) {
+        activeActivity = activity
     }
 
-    override fun onConversionDataSuccess(data: Map<String, Any>?) {
-        data?.map { (key, value) ->
-            Timber.tag(TAG).d("AppsFlyer onConversionDataSuccess Attribute: %s = %s", key, value)
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) = Unit
+    override fun onActivityStarted(activity: Activity) = Unit
```

Comment:
I've seen this ```= Unit``` in a couple of different places. What is ```Unit```?

Reply:
in Kotlin it's the equivalent of `void`

Reply:
another way to define this function would be:
```kotlin
override fun onActivityStarted(activity: Activity) {
}
```

using the `= Unit` here is basically saying there is no function body for this override.